From c2967bb7803e84e5dc7e34536caac7d389bfbcfd Mon Sep 17 00:00:00 2001
From: Ben Wolsieffer <benwolsieffer@gmail.com>
Date: Thu, 28 Nov 2019 00:07:08 -0500
Subject: [PATCH] [BACKPORT] cmd: pxe: Increase maximum path length

On NixOS, cross compiled kernels have long suffixes that cause them to
exceed the current maximum path length. The PXE/TFTP max path length is
used for extlinux.conf support as well, which is where this problem
usually manifest's itself.

Signed-off-by: Ben Wolsieffer <benwolsieffer@gmail.com>
Acked-by: Joe Hershberger <joe.hershberger@ni.com>
(cherry picked from commit 2c4e067d493d031d1267eea1c635d1eab93ce8f7)
---
 cmd/pxe.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmd/pxe.c b/cmd/pxe.c
index cce48e01e82..46e416818ac 100644
--- a/cmd/pxe.c
+++ b/cmd/pxe.c
@@ -21,7 +21,7 @@
 #include "menu.h"
 #include "cli.h"
 
-#define MAX_TFTP_PATH_LEN 127
+#define MAX_TFTP_PATH_LEN 512
 
 const char *pxe_default_paths[] = {
 #ifdef CONFIG_SYS_SOC
-- 
2.35.1

